tests/boundaries/BoundaryHandlerTest.cpp

```diff
@@ -0,0 +1,226 @@
+//
+// Created by U-jin Hong on 04.06.24.
+//
+
+#include <gtest/gtest.h>
+#include <fstream>
+#include <sstream>
+
+#include "boundaries/BoundaryHandler.h"
+#include "calculators/LJCalculator.h"
+#include "io/in/FileReader.h"
+
+// Helper function to compare two std::array<double, 3> with a given tolerance
+bool arraysNear(const std::array<double, 3>& arr1, const std::array<double, 3>& arr2, double tolerance) {
+    for (size_t i = 0; i < arr1.size(); ++i) {
+        if (std::fabs(arr1[i] - arr2[i]) > tolerance) {
+            return false;
+        }
+    }
+    return true;
+}
+
+// Custom Google Test macro for comparing std::array<double, 3>
+#define ASSERT_ARRAYS_NEAR(arr1, arr2, tolerance) \
+    ASSERT_PRED3(arraysNear, arr1, arr2, tolerance)
+
+namespace boundaries {
+// Testing Particles moving against every reflecting boundary
+// Check if the forces after reflection are correct
+    TEST(BoundaryHandlerTest, BasicReflection) {
+        SPDLOG_INFO("Testing Reflections...");
+        std::map<boundaries::BoundaryDirection, boundaries::BoundaryType> boundaryMap{};
+        std::array<double, 2> domain = {20.0, 20.0};
+        std::unique_ptr<calculators::Calculator> calculator = std::make_unique<calculators::LJCalculator>(1, 5, 3.0);
+        boundaries::BoundaryProperties properties{domain, boundaryMap};
+        // Initialization of BoundaryHandler
+        boundaries::BoundaryHandler handler{properties, calculator.get(), 1.0};
+        std::ofstream outfile("particles_near_reflecting_boundary.txt");
+        outfile << "0\n"  // Data code for Particles
+                << "4\n"  // Number of data sets
+                << "2\n"  // Dimension of simulation
+                << "10.0 0.5 0.0     0.0 3.0 0.0      1.0\n" // Bottom
+                << "10.0 19.5 0.0    0.0 3.0 0.0      1.0\n" // Top
+                << "0.5 10.0 0.0     -3.0 0.0 0.0     1.0\n" // Left
+                << "19.5 10.0 0.0    3.0 0.0 0.0      1.0\n";// Right
+        outfile.close();
+
+        // Read cuboid data from the file
+        ParticleContainer container = FileReader::readFile("particles_near_reflecting_boundary.txt");
+        std::remove("particles_near_reflecting_boundary");
```

Comment:
Don't add the Particles via the FileReader in the tests. Just create a vector of particles and add them to the container.  
We want to test features in isolation as best as possible.